pontoon/sync/vcs/repositories.py

```diff
@@ -357,10 +357,14 @@ def revision(self):
         )
         return output.strip() if code == 0 else None
 
+    def _strip(self, rev):
+        "Ignore trailing + in revision number. It marks local changes."
+        return rev.rstrip('+')
+
     def get_changed_files(self, path, from_revision, statuses=None):
         statuses = statuses or ('A', 'M')
         code, output, error = self.execute(
-            ['hg', 'status', '-a', '-m', '-r', '--rev={}'.format(from_revision), '--rev=tip'],
+            ['hg', 'status', '-a', '-m', '-r', '--rev={}'.format(self._strip(from_revision)), '--rev=tip'],
```

Comment:
I agree, but I'll leave it for now in case we need that `+` bit later for some reason.